torchvision/datasets/omniglot.py

```diff
@@ -0,0 +1,195 @@
+from __future__ import print_function
+from PIL import Image
+import os
+import random
+import torch.utils.data as data
+from .utils import download_url, check_integrity, list_dir, list_files
+
+
+class Omniglot(data.Dataset):
+    """`Omniglot <https://github.com/brendenlake/omniglot>`_ Dataset.
+    Args:
+        root (string): Root directory of dataset where directory
+            ``omniglot-py`` exists.
+        background (bool, optional): If True, creates dataset from the "background" set, otherwise
+            creates from the "evaluation" set. This terminology is defined by the authors.
+        transform (callable, optional): A function/transform that  takes in an PIL image
+            and returns a transformed version. E.g, ``transforms.RandomCrop``
+        target_transform (callable, optional): A function/transform that takes in the
+            target and transforms it.
+        download (bool, optional): If true, downloads the dataset zip files from the internet and
+            puts it in root directory. If the zip files are already downloaded, they are not
+            downloaded again.
+    """
+    folder = 'omniglot-py'
+    download_url_prefix = 'https://github.com/brendenlake/omniglot/raw/master/python'
+    zips_md5 = {
+        'images_background': '68d2efa1b9178cc56df9314c21c6e718',
+        'images_evaluation': '6b91aef0f799c5bb55b94e3f2daec811'
+    }
+
+    def __init__(self, root, background=True,
+                 transform=None, target_transform=None,
+                 download=False):
+        self.root = os.path.join(os.path.expanduser(root), self.folder)
+        self.background = background
+        self.transform = transform
+        self.target_transform = target_transform
+
+        if download:
+            self.download()
+
+        if not self._check_integrity():
+            raise RuntimeError('Dataset not found or corrupted.' +
+                               ' You can use download=True to download it')
+
+        self.target_folder = os.path.join(self.root, self._get_target_folder())
+        self._alphabets = list_dir(self.target_folder)
+        self._characters = sum(
+            [
+                [
+                    os.path.join(alphabet, character)
+                    for character in list_dir(os.path.join(self.target_folder, alphabet))
+                ]
+                for alphabet in self._alphabets
+            ],
+            []
+        )
+        self._character_images = [
```

Comment:
This doesn't look like it's used in this class, why not just wrap this with the sum and use it everywhere you use `_flat_character_images`. You can probably also make this span a few less lines by bunching the `[[` etc.

Reply:
I wanted to have an easy way to sample images from a given kind of character id for the randomized pair generation. I discuss this more in the other comment.